### PR TITLE
[lodash] clean up _.sample typings

### PR DESCRIPTION
**Pull request by bcherny**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://lodash.com/docs/4.17.4#sample
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/lodash/index.d.ts [hunk 1]
```diff
@@ -9232,21 +9232,7 @@ declare namespace _ {
...
-        sample<O extends Object, T>(
-            collection: O | null | undefined
-        ): T | undefined;
-
-        /**
-         * @see _.sample
-         */
-        sample<T>(
-            collection: Object | null | undefined
+            collection: List<T> | Dictionary<T> | NumericDictionary<T> | object | string | null | undefined
```
Comment (aj-r) [inline types/lodash/index.d.ts]: I would remove `string` from the list of types here. `string` is assignable to `List<string>`, so adding `string` is redundant. Plus, using `List<string>` will cause `T` to be correctly inferred as a string.

**Comment by aj-r** [comment]
I like it, just want to see one minor fix

File: types/lodash/index.d.ts (hunk 1 above)
Comment (bcherny) [inline types/lodash/index.d.ts]: I didn't know that - good catch!

**Review by aj-r — APPROVED** [review]

**Review by stepancar — APPROVED** [review]

**Comment by ericanderson** [comment]
:+1:

**Comment by bcherny** [comment]
@aj-r Do you know how to placate @typescript-bot? Maybe it's enough to respond to your comment.

**Review by ericanderson — APPROVED** [review]

**Comment by aj-r** [comment]
I tried deleting my comment on the code - hopefully that works?

**Pull request merged**